scripts/setup/requirements.esp32.txt

```diff
@@ -2,13 +2,13 @@ click>=7.0
 future>=0.15.2
 pyparsing>=2.0.3,<2.4.0
 idf-component-manager
-pygdbmi<=0.9.0.2
+pygdbmi<=0.9.0.2 ; python_version < "3.11"
 reedsolo>=1.5.3,<=1.5.4
 bitstring>=3.1.6
 ecdsa>=0.16.0
 kconfiglib==13.7.1
 construct==2.10.54
-python-socketio<5
-gdbgui==0.13.2.0 ; platform_machine != 'aarch64' and sys_platform == 'linux'
-jinja2<3.1
-itsdangerous<2.1
+python-socketio<5 ; python_version < "3.11"
+gdbgui==0.13.2.0 ; platform_machine != 'aarch64' and sys_platform == 'linux' and python_version < "3.11"
```

Comment:
Isn't `gdbgui` needed for `idf.py build`?

Reply:
@bzbarsky-apple `gdbgui` is needed for `idf.py build`

Reply:
> @bzbarsky-apple `gdbgui` is needed for `idf.py build`

Not sure about that, espressif-idf removed it as well:

see: https://github.com/espressif/esp-idf/commit/2685a0a203846f0260b793673ff3ddcce1e12cde
and: https://github.com/espressif/esp-idf/commit/a816dfbfc3a9723db3154884a5e7d5a6f1853314

as well as the issue here: https://github.com/espressif/esp-idf/issues/10116

i think it is only used, when you run `idf.py gdbgui` which won´t work with python 3.11 as gdbgui does not support Python 3.11+

Reply:
The issue with older esp is that idf.py validates the environment, specifically when we do a `cmake generate`. So even though we would compile just fine, we fail cmake at start because gdbui is missing :( 

We did complain to ESP and were told that indeed this was removed in 5.0 I believe ... still waiting to get matter updated to the latest SDK and hoping then this will go away.

Reply:
My vote would be to accept this because it seems to allow more builds (all except ESP) on newer pythons where as without this we allow no builds.

The downside is that errors would be harder to debug (we would get esp builds saying "missing gdbui" or maybe something more cryptic) during cmake generate. However this will fix itself once we update to the newest Espressif SDK.

@dhrishi do we happen to know a timeline for a SDK update?

Reply:
@andy31415 IDF 5.0 is currently supported with connectedhomeip and examples should compile (#24720 #24983). Although, we have not officially moved to it as there may be some customers who still wish to use IDF v4.4 and not change at the last moment.

Will accepting this patch result into build failure for ESP builds for v4.4.x with Python 11?

Reply:
With python 11 the build fails no matter what: without this patch it fails trying to install gdbui, with this patch it will fail at `idf.py build` time due to lack of gdbui.

Reply:
This patch however allows bootstrap and compile for other environments ... so while esp32 will fail on py11 no matter what, without this patch everything else fails, with this patch others may pass (e.g. linux builds or other platforms that are not esp)

Reply:
I think ideally we would have something that very clearly fails with an actionable error (e.g. "python 3.11 is not supported") in the cases when we really can't compile with it, while compiling in other cases.  Our current errors are completely inscrutable.

If we can't do that, I agree that this is probably better than nothing....

Reply:
@dhrishi - what should we do here? as this does not make things worse, it seems it should be acceptaable.